### DISCUSSION TITLE
自搜狂魔上线

### DISCUSSION THREAD

**Discussion by kengwang** [HyPlayer/HyPlayer#40] [General]
![image](https://user-images.githubusercontent.com/30862240/120968957-271d7680-c79c-11eb-8ce2-8e5e5568b21f.png)
原文地址: https://www.pxr0.com/4514.html
贴一下评价原话
> HyPlayer 和 LyricEase 很像，但可惜只是个初出茅庐的新人，连作者都承认 HyPlayer 还在刚开始的开发阶段。
> 比如推荐歌单没办法二次筛选，没有歌曲分类，也没有个人用户的歌单供你选择（所以对于淘歌来说，HyPlayer 不是一个好去处）。
> 比如当前本地文件格式只支持 MP3 和 Flac 两种等等。
> 既然有这些地方不完善，为啥我还要推荐 HyPlayer，其实原因就一个，HyPlayer 能获取我调教了许久的官方每日推荐歌单。
> 
> 除此之外，我个人创建或收藏的歌单 HyPlayer 也能完美获取，对于我这种已经半只脚迈出网易云音乐的用户而言，HyPlayer 完美解决了我的需求：
> 
> 官方每日推荐和已收藏创建的歌单（需求第一位）。
> 
> 虽然 HyPlayer 还是个新人，但每天用它听几首推荐歌曲没一点问题，我对它唯一的期望就是像 YesPlayMusic 一样多搞一些其他平台的接口，解锁一下全平台搜索的功能，除了这些我对 HyPlayer 就没啥挑剔的了

在后面的楼层会对一些东西作出解释

**Comment by kengwang** [comment]
1. 和LyricEase很像
天啊,太高看我们的UI了吧. 我觉得现在的UI特别简陋,不是一般的简陋. 而LyricEase 的UI十分有设计感. 毕竟HyPlayer大家都可以看到最早的提交日期是 2021/2/15 , 也就是还有几天开学的时间. 开发之前完全不知道有LyricEase,不然可能就不会有HyPlayer了. 开发倒是和YesPlayMusic有关,当时这个播放器很火,所有就安装了. 不过发现居然是Electron! (我对Electron有种天然的排斥感) 最令我不适的是在播放前有一段空白期,我也不知道他在干嘛... .... 所有我先单干开始用 `.NET` 写播放器,开始是准备.NET Framework + WPF 的,也当是WPF的练手,期初也不叫HyPlayer 而叫NCPlayer 图标也是我再iconfont随便找的一个. 在某一次契机下找 @aaaaaaccd 的时候,他问我为什么不用UWP呢? 于是乎我就走进了UWP这个坑... ...
2. 关于完成度
到目前为止,HyPlayer也还处于快速迭代期,功能还在利用每周周末和放假的时间来慢慢补,偶尔也会有社区大佬Pr. 某些功能我个人不常用,所以都一直没有做.
3. 关于其他的功能性问题
有些已经开发出来了,还有的正在咕咕咕
4. 本地格式
还需要支持什么格式呢? 我这边下载下来的都是 mp3 和 flac,偶尔有些番的BD里面附赠点wav
5. 其他平台接口
这个功能不会搞. 因为 UnblockNeteaseMusic 项目本身就是Node.js 写的,所以YesPlay才那么顺手就可以拿过来用了. 我也暂时不会进行 C# 的移植,因为可能会涉及到版权问题或者其他杂七杂八的东西

**Reply by EP012014** [reply]
加油，我们一起奋斗，高举UWP大旗！
